frontend/src/api/k8s/__tests__/configMaps.spec.ts

```diff
@@ -0,0 +1,244 @@
+  it('should successfully create a configMap', async () => {
+    k8sCreateResourceMock.mockResolvedValue(configMapMock);
+
+    const result = await createConfigMap(
+      assembleConfigMap(projectName, configMapData, configMapName),
+    );
+    expect(k8sCreateResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sCreateResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: {
+        apiVersion: 'v1',
+        kind: 'ConfigMap',
+        plural: 'configmaps',
+      },
+      queryOptions: { queryParams: {} },
+      resource: configMapMock,
+    });
+    expect(result).toStrictEqual(configMapMock);
+  });
+
+  it('should handle errors and rethrow', async () => {
+    k8sCreateResourceMock.mockRejectedValue(new Error('error1'));
+
+    await expect(
+      createConfigMap(assembleConfigMap(projectName, configMapData, configMapName)),
+    ).rejects.toThrow('error1');
+    expect(k8sCreateResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sCreateResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: {
+        apiVersion: 'v1',
+        kind: 'ConfigMap',
+        plural: 'configmaps',
+      },
+      queryOptions: { queryParams: {} },
+      resource: configMapMock,
+    });
+  });
+});
+
+describe('replaceConfigMap', () => {
+  it('should successfully replace configMap', async () => {
+    k8sUpdateResourceMock.mockResolvedValue(configMapMock);
+
+    const result = await replaceConfigMap(
+      assembleConfigMap(projectName, configMapData, configMapName),
+    );
+    expect(k8sUpdateResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sUpdateResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: {
+        apiVersion: 'v1',
+        kind: 'ConfigMap',
+        plural: 'configmaps',
+      },
+      queryOptions: { queryParams: {} },
+      resource: configMapMock,
+    });
+    expect(result).toStrictEqual(configMapMock);
+  });
+
+  it('should handle errors and rethrow', async () => {
+    k8sUpdateResourceMock.mockRejectedValue(new Error('error1'));
+
+    await expect(
+      replaceConfigMap(assembleConfigMap(projectName, configMapData, configMapName)),
+    ).rejects.toThrow('error1');
+    expect(k8sUpdateResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sUpdateResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: {
+        apiVersion: 'v1',
+        kind: 'ConfigMap',
+        plural: 'configmaps',
+      },
+      queryOptions: { queryParams: {} },
+      resource: configMapMock,
+    });
+  });
+});
+
+describe('deleteConfigMap', () => {
+  it('should successfully delete configMap', async () => {
+    const mockK8sStatus: K8sStatus = {
+      apiVersion: 'v1',
+      kind: 'Status',
+      code: 200,
+      message: '200 OK',
+      reason: '',
+      status: 'Success',
+    };
+    k8sDeleteResourceMock.mockResolvedValue(mockK8sStatus);
```

Comment:
Can we make use of a mock function to create this.

Reply:
We dont have  a mock function for success  k8status

Reply:
I have created a mock for 200 status in this PR: https://github.com/opendatahub-io/odh-dashboard/pull/2366. If that gets approved I think you can re-use it here.

Reply:
Done